agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -482,179 +455,221 @@ protected static void replace_instructions(
-   * All method names and argument types should be fully-qualified.
+   * <p>Blank lines and // comments are ignored. A call to a method in a class of the
+   * orig-package-name is replaced by a call to the same method within the new-package-name. All
+   * method names and argument types should be fully-qualified.
    *
-   * @param map_file the file with map of method substitutions
-   * @throws IOException if file has missing regex
+   * @param in the {@code Reader} for the replacement file
+   * @throws IOException if there is an error reading from the file
    */
-  public void read_map_file(File map_file) throws IOException {
-
-    LineNumberReader lr = new LineNumberReader(new FileReader(map_file));
-    MapFileErrorHandler mfeh = new MapFileErrorHandler(lr, map_file);
-    Pattern current_regex = null;
-    Map<MethodDef, ReplacementClass> map = new LinkedHashMap<>();
+  void readMapFile(Reader in) throws IOException {
+    LineNumberReader lr = new LineNumberReader(in);
     for (String line = lr.readLine(); line != null; line = lr.readLine()) {
-      line = line.replaceFirst("//.*$", "");
-      if (line.trim().length() == 0) continue;
-      if (line.startsWith(" ")) {
-        if (current_regex == null)
-          throw new IOException("No current class regex on line " + lr.getLineNumber());
-        StrTok st = new StrTok(line, mfeh);
-        st.stok.wordChars('.', '.'); // make '.' a word constituent
-        MethodDef md = parse_method(st);
-        String replacement_method_class = st.need_word(); // words including '.'
-        map.put(md, new ReplacementClass(replacement_method_class));
+      line = line.replaceFirst("//.*$", "").trim();
+      if (line.length() == 0) {
+        continue;
+      }
+      Matcher sigMatcher = SIGNATURE_PATTERN.matcher(line);
+      if (sigMatcher.find() && sigMatcher.groupCount() == 2) {
+        String[] arguments = new String[0];
+        String argString = sigMatcher.group(2);
+        if (!argString.isEmpty()) {
+          arguments = argString.split(",");
+        }
+        MethodDef orig = getMethod(sigMatcher.group(1).trim(), arguments);
+        if (sigMatcher.find() && sigMatcher.groupCount() == 2) {
+          arguments = sigMatcher.group(2).split(",");
+          MethodDef replacement = getMethod(sigMatcher.group(1).trim(), arguments);
+          replacementMap.put(orig, replacement);
+        }
       } else {
-        if (current_regex != null) {
-          MethodMapInfo mmi = new MethodMapInfo(current_regex, map);
-          map_calls_file.add(mmi);
-          map = new LinkedHashMap<>();
+        Matcher prefixMatcher = PREFIX_PATTERN.matcher(line);
+        if (prefixMatcher.find()) {
+          String orig = prefixMatcher.group(0).trim();
+          if (prefixMatcher.find()) {
+            prefixReplacementMap.put(orig, prefixMatcher.group(0).trim());
+          }
         }
-        current_regex = Pattern.compile(line);
       }
```

Comment:
If a line is not matched, this simply ignores it.  That will be very confusing for users who get a single character wrong in their file.  If a line is not matched, the program should report that to the user and halt.